### PR TITLE
RR-775 - Initial GET handler for Hoping To Work On Release

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR start to build out the Create Induction journey, but implementing the basic GET route for the very first page in the journey (Do they want to get work on release?)

The approach is to write controllers that extend the base classes for each page.
EG we already have the following hierarchy:
* `abstract class InductionController`
Contains common functions that all Induction related controllers will either inherit or must implement

  * `abstract class HopingToWorkOnReleaseController extends InductionController`
  Contains the GET handler function for Hoping To Work On Release that will work for both Updating and Creating an Induction
  
    * `class HopingToWorkOnReleaseUpdateController extends HopingToWorkOnReleaseController`
    Contains the POST handler for Hoping To Work On Release for Updating an Induction
    
Following this pattern, this PR adds:
* `class HopingToWorkOnReleaseCreateController extends HopingToWorkOnReleaseController`
Currently implements the methods from the abstract superclasses, but in my next PR we will add the POST handler for Hoping To Work On Release for Creating an Induction

As well as defining the initial controller, this PR starts defining the routes for creating an Induction, with the route:
`GET /prisoners/:prisonNumber/create-induction/hoping-to-work-on-release` and updates the anchor links in the various nunjucks templates that allow the user to start the process of creating an induction.

It also updates a cypress test that previously worked on the assumption that the Induction UI was in the CIAG UI rather than pages that could be rendered by this project.

### CODE REVIEW

File: integration_tests/e2e/overview/preInductionOverview.cy.ts [hunk 1]
```diff
@@ -100,16 +77,12 @@ context('Prisoner Overview page - Pre Induction', () => {
     overviewPage.clickMakeProgressPlan()
 
     // Then
-    Page.verifyOnPage(CreateCiagInductionPage)
+    Page.verifyOnPage(HopingToWorkOnReleasePage) //
```
Comment (nathanrussell-moj-digital) [inline integration_tests/e2e/overview/preInductionOverview.cy.ts:80]: This is the crux of the change in this cypress test in respect of the new page implementation

File: integration_tests/e2e/overview/preInductionOverview.cy.ts [hunk 2]
```diff
@@ -1,36 +1,20 @@
...
-    cy.task('stubSignInAsUserWithEditAuthority')
-    cy.task('stubAuthUser')
-    cy.task('stubGetHeaderComponent')
-    cy.task('stubGetFooterComponent')
-    cy.task('stubPrisonerList')
-    cy.task('stubCiagInductionList')
-    cy.task('stubActionPlansList')
-    cy.task('getPrisonerById', 'G6115VJ')
-    cy.task('getPrisonerById', 'A00001A')
+    cy.signInAsUserWithEditAuthorityToArriveOnPrisonerListPage()
```
Comment (nathanrussell-moj-digital) [inline integration_tests/e2e/overview/preInductionOverview.cy.ts:11]: Simplified the beforeEach hook by using the new `signInAsUserWithEditAuthorityToArriveOnPrisonerListPage` cypress command that I added the other day (the new command contains most of tasks that we were individually setting up beforehand)

File: integration_tests/e2e/overview/preInductionOverview.cy.ts [hunk 3]
```diff
@@ -65,29 +45,26 @@ context('Prisoner Overview page - Pre Induction', () => {
 
   it('should render prisoner Overview page listing no prisoner goals given prisoner has no goals created pre-induction', () => {
     // Given
-    cy.task('stubSignInAsUserWithEditAuthority')
-    cy.signIn()
-
-    const prisonNumber = 'A00001A'
+    const prisonNumberForPrisonerWithNoGoals = 'A00001A'
```
Comment (nathanrussell-moj-digital) [inline integration_tests/e2e/overview/preInductionOverview.cy.ts:48]: This test uses a different prisoner that is setup with different stub data, hence having to re-setup the stubs

File: server/routes/induction/common/hopingToWorkOnReleaseController.ts [hunk 4]
```diff
@@ -39,6 +39,6 @@ export default abstract class HopingToWorkOnReleaseController extends InductionC
 
 const toHopingToWorkOnReleaseForm = (inductionDto: InductionDto): HopingToWorkOnReleaseForm => {
   return {
-    hopingToGetWork: inductionDto.workOnRelease?.hopingToWork,
+    hopingToGetWork: inductionDto?.workOnRelease?.hopingToWork,
```
Comment (nathanrussell-moj-digital) [inline server/routes/induction/common/hopingToWorkOnReleaseController.ts:42]: Minor change here - in the case of creating an Induction the `InductionDto` that we get from the session is undefined.
Putting the `?` here is the right thing to do as it means the resultant form field `hopingToGetWork` is undefined which means the user will have to answer it (when we implement the POST submit handler and validation)

File: server/routes/induction/create/hopingToWorkOnReleaseCreateController.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+export default class HopingToWorkOnReleaseCreateController extends HopingToWorkOnReleaseController {
+  getBackLinkUrl(req: Request): string {
+    const { prisonNumber } = req.params
+    return `/plan/${prisonNumber}/view/overview`
+  }
+
+  getBackLinkAriaText(req: Request): string {
+    return getDynamicBackLinkAriaText(req, this.getBackLinkUrl(req))
+  }
+}
```
Comment (nathanrussell-moj-digital) [inline server/routes/induction/create/hopingToWorkOnReleaseCreateController.ts:14]: We will add the submit handler method to this class in the next PR

File: server/routes/induction/create/index.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    ])
+    router.post('/prisoners/:prisonNumber/create-induction/**', [
+      checkUserHasEditAuthority(),
+      retrievePrisonerSummaryIfNotInSession(services.prisonerSearchService),
+    ])
+
+    router.get('/prisoners/:prisonNumber/create-induction/hoping-to-work-on-release', [
+      hopingToWorkOnReleaseCreateController.getHopingToWorkOnReleaseView,
+    ])
+  }
```
Comment (nathanrussell-moj-digital) [inline server/routes/induction/create/index.ts:30]: We will add the `router.post` call for the submit handler in the next PR

File: server/views/pages/overview/partials/educationAndTrainingTab/educationAndTrainingTabContents.njk [hunk 7]
```diff
@@ -40,9 +40,17 @@ The 'Training and eduction in prison' summary card is only shown if the short qu
           </div>
 
           <div class="govuk-summary-card__content">
+            {%- set createInductionUrl -%}
```
Comment (nathanrussell-moj-digital) [inline server/views/pages/overview/partials/educationAndTrainingTab/educationAndTrainingTabContents.njk:43]: The following changes all set the URL of the Create Induction journey to either the PLP or CIAG UI first page, depending on whether the feature toggle is set

**Review by leejacobson-moj — APPROVED** [review]
Looks good. Approved. 👍

**Pull request merged**